controllers/apim/ratelimitpolicy_controller.go

```diff
@@ -380,6 +385,58 @@ func (r *RateLimitPolicyReconciler) gatewayRefListFromHTTPRoute(httpRoute *gatew
 	return gwKeys
 }
 
+func (r *RateLimitPolicyReconciler) validateRuleHosts(ctx context.Context, rlp *apimv1alpha1.RateLimitPolicy) error {
+	var netObj client.Object
+	var err error
+
+	if rlp.IsForGateway() {
+		netObj, err = r.fetchGateway(ctx, rlp)
+		if err != nil {
+			// The object should also exist
+			return err
+		}
+	} else if rlp.IsForHTTPRoute() {
+		netObj, err = r.fetchHTTPRoute(ctx, rlp)
+		if err != nil {
+			// The object should also exist
+			return err
+		}
+	} else {
+		return errors.New("validateRuleHosts: rlp targeting unknown network resource")
+	}
```

Comment:
Make sense to move this to another function? Maybe here, maybe a func of `apimv1alpha1.RateLimitPolicy`?

Reply:
this will add new changes to the PR, but since you asked :rocket:

Reply:
Done in https://github.com/Kuadrant/kuadrant-controller/pull/212